components/unarchive_channel_modal/unarchive_channel_modal.jsx

```diff
@@ -0,0 +1,126 @@
+
+export default class UnarchiveChannelModal extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        * Function called when modal is dismissed
+        */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * channel data
+         */
+        channel: PropTypes.object.isRequired,
+
+        canViewArchivedChannels: PropTypes.bool,
+
+        actions: PropTypes.shape({
+
+            /**
+            * Function called for deleting channel,
+            */
+
+            unarchiveChannel: PropTypes.func.isRequired,
+        }),
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {show: true};
+    }
+
+    handleUnarchive = () => {
+        if (this.props.channel.id.length !== Constants.CHANNEL_ID_LENGTH) {
+            return;
+        }
+        this.props.actions.unarchiveChannel(this.props.channel.id);
+        this.onHide();
+    }
+
+    onHide = () => {
+        this.setState({show: false});
+    }
+
+    render() {
+        const {canViewArchivedChannels} = this.props;
+        return (
+            <Modal
+                dialogClassName='a11y__modal'
+                show={this.state.show}
+                onHide={this.onHide}
+                onExited={this.props.onHide}
+                role='dialog'
+                aria-labelledby='unarchiveChannelModalLabel'
+                id='unarchiveChannelModal'
+            >
+                <Modal.Header closeButton={true}>
+                    <Modal.Title
+                        componentClass='h1'
+                        id='unarchiveChannelModalLabel'
+                    >
+                        <FormattedMessage
+                            id='unarchive_channel.confirm'
+                            defaultMessage='Confirm UNARCHIVE Channel'
+                        />
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <div className='alert alert-danger'>
+                        {!canViewArchivedChannels &&
+                            <FormattedMarkdownMessage
+                                id='unarchive_channel.question'
+                                defaultMessage='This will unarchive the channel from the team and make its contents accessible for all users. \n \nAre you sure you wish to unarchive the **{display_name}** channel?'
+                                values={{
+                                    display_name: this.props.channel.display_name,
+                                }}
+                            />}
+                        {canViewArchivedChannels &&
+                            <FormattedMarkdownMessage
+                                id='unarchive_channel.viewArchived.question'
+                                defaultMessage={'This will unarchive the channel from the team.\n \nAre you sure you wish to unarchive the **{display_name}** channel?'}
```

Comment:
@adamjclarkson Same thing here.
```suggestion
                                defaultMessage={'Are you sure you wish to unarchive the **{display_name}** channel?'}
```